src/scenegraph/Loader.cpp

```diff
@@ -982,4 +983,46 @@ namespace SceneGraph {
 		return 0x0;
 	}
 
+	void Loader::ParseGunTags(Model *m)
+	{
+		const std::string test = "tag_gunmount";
+		SceneGraph::Model::TVecMT mounts_founds;
+		m->FindTagsByStartOfName(test, mounts_founds);
+
+		m->m_mounts.clear();
+		m->m_mounts.reserve(mounts_founds.size());
+
+		bool break_; // <- Used to "break" from inner 'for' cycle
+		for (int i = 0; i < mounts_founds.size(); i++) {
+			break_ = false;
+			const std::string &name = mounts_founds[i]->GetName();
+			if (name.length() > 14) {
+				// Multiple "tag" type: we group tags with
+				// the same index
+				std::string name_to_first_index = name.substr(0,14);
+				for (int j = 0; j < m->m_mounts.size(); j++ )
```

Comment:
`O(n*(n-1))` recursion isn't exactly pretty. Have you considered using another container with a `count` method instead, or a hashmap?

Reply:
Done... I fear that it isn't any faster but at least is far more readable.

Reply:
Well, turns out that I used a "not existing" function which only works for my combination of S.O. & compiler & cabala & whatsoever... I will change it to a more common paradigma in few days. @Web-eWorks , in the meantime if you have other points you wish to see improved, let me know